### PR TITLE
Add samples and update docs for RangeFilter APIs

### LINKED ISSUE #2398: RangeFilter has no samples
The RangeFilter has no sample to show how to use it. I would like to see a sample and links to the sample from the catalog entries.

### PR DESCRIPTION
**Pull request by rogancarr**
This PR adds a sample for `FilterByColumn` and `FilterByKeyColumnFraction` and updates their docs.

Fixes #2398

### CODE REVIEW

File: src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs [hunk 1]
```diff
@@ -124,14 +124,25 @@ public class SampleTemperatureData
             public float Temperature { get; set; }
         }
 
-        public static IEnumerable<SampleTemperatureData> GetSampleTemperatureData()
+        /// <summary>
+        /// Get a fake temperature dataset.
+        /// </summary>
+        /// <param name="exampleCount">The number of examples to return.</param>
+        /// <returns>An enumerable of <see cref="SampleTemperatureData"/>.</returns>
+        public static IEnumerable<SampleTemperatureData> GetSampleTemperatureData(int exampleCount)
```
Comment (shmoradims) [inline src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs]: >int exampleCount [](start = 82, length = 16)

I suggest using 'yield return' for creating the lazy IEnumerable and removing the sampleCount argument. We're reinventing Take() in Linq, which does exactly this.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByKeyColumnFraction.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public static class FilterByKeyColumnFraction
+    {
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Get a small dataset as an IEnumerable.
+            IEnumerable<SamplesUtils.DatasetUtils.MulticlassClassificationExample> enumerableOfData = SamplesUtils.DatasetUtils.GenerateRandomMulticlassClassificationExamples(10);
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByKeyColumnFraction.cs]: >  [](start = 101, length = 1)

a line-break would help here. line's too long

File: src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs (hunk 1 above)
Comment (shmoradims) [reply src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs]: This is a style suggestion though. I'll leave it up to you.

---
In reply to: [253624396](https://github.com/dotnet/machinelearning/pull/2405#discussion_r253624396) [](ancestors = 253624396)

Comment (rogancarr) [reply src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs]: Good call on the `yield return`.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByKeyColumnFraction.cs (hunk 2 above)
Comment (rogancarr) [reply docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByKeyColumnFraction.cs]: Done.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByColumn.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class FilterByColumn
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByColumn.cs]: >FilterByColumn [](start = 24, length = 14)

Should there be commentary about what this example is doing?

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByKeyColumnFraction.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+using Microsoft.ML.SamplesUtils;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    using MulticlassClassificationExample = DatasetUtils.MulticlassClassificationExample;
+
+    public static class FilterByKeyColumnFraction
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByKeyColumnFraction.cs]: Same here - should there be any commentary about what this example is doing?

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByKeyColumnFraction.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            IEnumerable<MulticlassClassificationExample> enumerableOfData = DatasetUtils.GenerateRandomMulticlassClassificationExamples(10);
+            var data = mlContext.Data.ReadFromEnumerable(enumerableOfData);
+
+            // Convert the string labels to keys
+            var pipeline = mlContext.Transforms.Conversion.MapValueToKey("Label");
+            var transformedData = pipeline.Fit(data).Transform(data);
+
+            // Before we apply a filter, examine all the records in the dataset.
+            var enumerable = mlContext.CreateEnumerable<MulticlassWithKeyLabel>(transformedData, reuseRowObject: true);
+            Console.WriteLine($"Label\tFeatures");
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/FilterByKeyColumnFraction.cs]: >Label [](start = 32, length = 5)

I kind of like this here is a sample of the dataset and here is the transformed data. Im wondering if we should update all samples to do something along these lines.

**Pull request merged**